### PR TITLE
10.0 backport wechat

### PR DESCRIPTION
**Pull request by GabbasovDinar**

### CODE REVIEW

File: pos_qr_show/static/src/js/pos_qr_show.js [hunk 1]
```diff
@@ -76,4 +76,12 @@ odoo.define('pos_qr_show', function(require){
         },
     });
 
+    screens.PaymentScreenWidget.include({
+        show: function() {
+            // remove old QR code
+            this.pos.hide_payment_qr_on_payment_screen();
+            this._super();
+        }
+    });
```
Comment (yelizariev) [inline pos_qr_show/static/src/js/pos_qr_show.js:85]: :+1:

File: pos_wechat/static/src/js/wechat_pay.js [hunk 2]
```diff
@@ -97,10 +97,14 @@ odoo.define('pos_wechat', function(require){
     var OrderSuper = models.Order;
     models.Order = models.Order.extend({
         add_paymentline: function(cashregister){
-            if (cashregister.journal.wechat == 'native'){
+            if (cashregister.journal.wechat == 'native') {
+                // remove all payment lines (because the Wechat payment can pay only the entire order)
+                this.clean_empty_paymentlines();
```
Comment (yelizariev) [inline pos_wechat/static/src/js/wechat_pay.js]: Not sure about this

File: pos_wechat/static/src/js/wechat_pay.js [hunk 3]
```diff
@@ -97,10 +97,14 @@ odoo.define('pos_wechat', function(require){
...
                 this.pos.wechat_qr_payment(this, cashregister);
                 return;
             }
+            // hide QR code
+            this.pos.hide_payment_qr_on_payment_screen();
```
Comment (yelizariev) [inline pos_wechat/static/src/js/wechat_pay.js:105]: Seems ok

**Comment by yelizariev** [comment]
You shall not send commit that fixes previous commit of the same PR

**Comment by yelizariev** [comment]
At least when you use release tags

File: pos_wechat/static/src/js/wechat_pay.js [hunk 4]
```diff
@@ -97,10 +97,12 @@ odoo.define('pos_wechat', function(require){
     var OrderSuper = models.Order;
     models.Order = models.Order.extend({
         add_paymentline: function(cashregister){
-            if (cashregister.journal.wechat == 'native'){
+            if (cashregister.journal.wechat == 'native') {
```
Comment (yelizariev) [inline pos_wechat/static/src/js/wechat_pay.js]: wat

Comment (GabbasovDinar) [reply pos_wechat/static/src/js/wechat_pay.js]: @yelizariev 
247pac

**Pull request merged**